File: jelly-cinema-v2/im-debug/im_test.py
import os
import requests
import websocket

TOKEN = os.getenv("TOKEN", "PLEASE_SET_TOKEN")
BASE_HTTP = os.getenv("BASE_HTTP", "http://localhost:8080")
WS_URL = os.getenv("WS_URL", f"ws://localhost:8080/ws/chat?token={TOKEN}")


def test_http():
    headers = {"Authorization": TOKEN}

    print("\n=== HTTP: /im/sessions ===")
    r = requests.get(f"{BASE_HTTP}/im/sessions", headers=headers, timeout=8)
    print("status=", r.status_code)
    print(r.text)

    print("\n=== HTTP: /im/search/user?keyword=admin ===")
    r = requests.get(
        f"{BASE_HTTP}/im/search/user",
        headers=headers,
        params={"keyword": "admin"},
        timeout=8,
    )
    print("status=", r.status_code)
    print(r.text)


def test_ws():
    print("\n=== WS CONNECT ===")
    print("WS_URL=", WS_URL)

    def on_open(_ws):
        print("[ws] open")

    def on_message(_ws, message):
        print("[ws] message:", message)

    def on_error(_ws, error):
        print("[ws] error:", error)

    def on_close(_ws, close_status_code, close_msg):
        print(f"[ws] close: code={close_status_code}, reason={close_msg}")

    ws = websocket.WebSocketApp(
        WS_URL,
        header=[f"Authorization: {TOKEN}"],
        on_open=on_open,
        on_message=on_message,
        on_error=on_error,
        on_close=on_close,
    )

    ws.run_forever(ping_interval=10, ping_timeout=5)


if __name__ == "__main__":
    if not TOKEN or TOKEN == "PLEASE_SET_TOKEN":
        raise SystemExit(
            "Missing TOKEN. Use:\n"
            "  PowerShell: $env:TOKEN=\"xxx\"; python .\\im_test.py\n"
            "  or set TOKEN in your environment."
        )

    test_http()
    test_ws()

File: jelly-cinema-v2/im-debug/im-test.js
const axios = require('axios')
const WebSocket = require('ws')

const TOKEN = process.env.TOKEN || 'PLEASE_SET_TOKEN'
const BASE_HTTP = process.env.BASE_HTTP || 'http://localhost:8080'
const WS_URL =
  process.env.WS_URL ||
  `ws://localhost:8080/ws/chat?token=${encodeURIComponent(TOKEN)}`

async function testHttp() {
  const client = axios.create({
    baseURL: BASE_HTTP,
    timeout: 8000,
    headers: {
      Authorization: TOKEN
    }
  })

  console.log('\n=== HTTP: /im/sessions ===')
  try {
    const r1 = await client.get('/im/sessions')
    console.log('status=', r1.status)
    console.log('body=', JSON.stringify(r1.data))
  } catch (e) {
    console.error('HTTP /im/sessions failed:', e?.response?.status, e?.message)
    if (e?.response?.data) console.error('resp.data=', e.response.data)
  }

  console.log('\n=== HTTP: /im/search/user?keyword=admin ===')
  try {
    const r2 = await client.get('/im/search/user', { params: { keyword: 'admin' } })
    console.log('status=', r2.status)
    console.log('body=', JSON.stringify(r2.data))
  } catch (e) {
    console.error('HTTP /im/search/user failed:', e?.response?.status, e?.message)
    if (e?.response?.data) console.error('resp.data=', e.response.data)
  }
}

function testWs() {
  console.log('\n=== WS CONNECT ===')
  console.log('WS_URL=', WS_URL)

  const ws = new WebSocket(WS_URL, {
    headers: {
      Authorization: TOKEN
    }
  })

  const heartbeat = setInterval(() => {
    if (ws.readyState === WebSocket.OPEN) {
      ws.ping()
    }
  }, 10000)

  ws.on('open', () => {
    console.log('[ws] open')
  })

  ws.on('message', (data) => {
    console.log('[ws] message:', data.toString())
  })

  ws.on('pong', () => {
    console.log('[ws] pong')
  })

  ws.on('close', (code, reason) => {
    console.log(`[ws] close: code=${code}, reason=${reason ? reason.toString() : '(empty)'}`)
    clearInterval(heartbeat)
  })

  ws.on('error', (err) => {
    console.log('[ws] error:', err.message)
  })
}

async function main() {
  if (!TOKEN || TOKEN === 'PLEASE_SET_TOKEN') {
    console.error(
      'Missing TOKEN. Use:\n' +
        '  PowerShell: $env:TOKEN="xxx"; npm run test:im\n' +
        '  or set TOKEN in your environment.'
    )
    process.exit(1)
  }

  await testHttp()
  testWs()
}

main()
